torch/_dynamo/eval_frame.py

```diff
@@ -837,12 +837,15 @@ def graph_with_interpreter(*args):
                 break
 
         with enable_python_dispatcher(), fake_tensor_mode:
-            graph = make_fx(
-                graph_with_interpreter,
-                decomposition_table=decomposition_table,
-                tracing_mode="real",
-                _allow_non_fake_inputs=True,
-            )(*example_fake_inputs)
+            try:
+                graph = make_fx(
+                    graph_with_interpreter,
+                    decomposition_table=decomposition_table,
+                    tracing_mode="real",
+                    _allow_non_fake_inputs=True,
+                )(*example_fake_inputs)
+            except TypeError as e:
```

Comment:
This looks wrong. You have no reason to believe the TypeError is specifically because of dynamic control flow. Classify the errors at raise site, not here.

Reply:
Addressed by defining and catching cond specific exceptions